[Feature] Checkbox/CheckboxGroup: Add heading role and list position

narin opened the pull request:
<!-- PR title naming convention:
'[Issue type] Brief summary of issue suitable for changelog or copy/paste issue title',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

<!-- Preferred branch naming convention:
'[Issue type]/[Issue #]-[Your name]-[Summary of issue]',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

## Description of Change
<!-- Describe the change and context with which it was made beyond ACs unless straightforward.
Consider:
 - What is relevant to code reviewer(s) and not in the ticket?
 - What context may be relevant to a future dev or you in 6 months about this PR?
 - Did the course of work lead to notable dead ends? If so, why didn't they pan out?
 - Did the change add new dependencies? Why?
 - Were there important sources to link? Examples: an open bug with a dependency project, an article of someone else solving the same problem that was partially or wholly copied, external documentation relevant to solution
 -->
This PR contains accessibility improvements as suggested by @brea11y.
- Adds `heading` accessibility role to Header when provided to Checkbox or Checkbox group
- Add convenience hook `useA11yListPosition` to return `listPosition` translation when provided position and total length
  - Updated SegmentedControl to use this hook
- Adds `accessibilityValue` to Checkbox items within Checkbox group to allow screen reader to read out position within list

#### Testing Packages
<!-- List or range of alpha/beta packages published in association with this PR, if any -->


### Screenshots/Video
<!-- Add screenshots or video as needed; before/after recommended if appropriate. 
Convenience side-by-side formatting:
Before/after: <img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" />
Accordion before/after: <details><summary>Before/after</summary><img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" /></details>
-->

https://github.com/user-attachments/assets/d2766d7a-6758-422f-b5c4-4dd51228d46d


#### A11y label videos

https://github.com/user-attachments/assets/c4d3a27d-f279-47a5-8eb9-5d84b6c5a19b

https://github.com/user-attachments/assets/d615f978-70ac-4d50-bc18-41e1222d03e7





## Testing
<!-- Describe testing conducted to validate changes.
Consider highlighting:
- What testing was not explicitly done and may be relevant for QA? 
- Edge cases validated
- Special situations that could not be tested
- Any testing performed in a consuming app -->

- [x] Tested on iOS <!-- simulator is fine -->
- [x] Tested on Android <!-- simulator is fine -->
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish
<!-- Most changes entail a version increment; section can be removed for PRs exclusively within non-ship-relevant files (e.g. unit tests, Storybook stories) -->
If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/documentation/versioning.md) and the PR is approved and ready to merge:
- [ ] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

TimRoe commented on packages/components/src/components/Checkbox/Checkbox.tsx:
```diff
@@ -130,6 +131,7 @@ export const Checkbox: FC<CheckboxProps> = ({
         <Pressable
           onPress={onPress}
           style={tile ? tileStyle : pressableBaseStyle}
+          accessibilityValue={accessibilityValue}
```
Could be updated to `aria-valuetext` to align with newer/more web-like RN props.

TimRoe replied:
Nvm, couldn't here using the more general RN typing without updating that to string.

TimRoe commented on packages/components/src/utils/hooks/useA11yListPosition.ts:
```diff
@@ -0,0 +1,17 @@
+
+  return {
+    text: t('listPosition', {
+      position,
```
Should the +1 be baked into the hook? And rename the argument from `position` to `index` to set the expectation it counts from 0.

For user facing counting, the expectation is it'd never start from 0 so seems reasonable to me to roll that into the utility and only correct as a one-off with -1 if for some reason we were working with values that didn't start from 0.

TimRoe commented on packages/components/src/components/CheckboxGroup/CheckboxGroup.tsx:
```diff
@@ -142,10 +142,16 @@ export const CheckboxGroup: FC<CheckboxGroupProps> = ({
             <Fragment key={`checkbox-group-item-${index}`}>
               <Checkbox
                 label={item}
+                accessibilityValue={accessibilityValue}
```
Could be updated to `aria-valuetext` to align with newer/more web-like RN props.

TimRoe commented on packages/components/src/components/SegmentedControl/SegmentedControl.tsx:
```diff
@@ -83,12 +85,7 @@ export const SegmentedControl: FC<SegmentedControlProps> = ({
-        total: labels.length,
-      }),
-    }
+    const accessibilityValue = useA11yListPosition(index + 1, labels.length)
```
Could potentially be in-line without wrapping now, eliminating the variable.

TimRoe commented on packages/components/src/components/Checkbox/Checkbox.tsx:
```diff
@@ -130,6 +131,7 @@ export const Checkbox: FC<CheckboxProps> = ({
         <Pressable
```
Does it cause issues with this being on the Pressable? Namely, do the a11yLabel overrides still work correctly when they're now nested a layer lower than the a11yValue? The PR video only highlights options 3/4 which don't have a label override.

narin replied:
Confirmed that the a11y label overrides still work correctly with voiceover

TimRoe reviewed (changes requested):
Mainly requesting changes as I think these do warrant adding to the unit tests for Checkbox/CheckboxGroup (validating header role and a11yValue) and the general ticket for that has already been worked.

Additionally:
- Recommend moving the +1 into the `useA11yListPosition` hook
- Unclear if the a11yValue being at a different level from a11yLabel causes issues with both being read as expected based on PR video

Beyond that, some minor suggestions.

narin commented:
Ready for another review @TimRoe:
- Updated list position hook to accept index and add + 1 to calculate position
- Alphabetized exports in accessibility.ts
- Updated components to use hook inline instead of declaring separately
- Updated Checkbox and SegmentedControl to use `aria-valuetext` for list position

TimRoe reviewed (changes requested):
Changes look good, but noticed a thing I didn't before: since this is a hook, we should be instantiating it at the top with a `const a11yListPosition = useA11yListPosition()` or whatever to be certain we aren't conditionally changing the number of hooks, right? I don't think is an issue to these particular use cases, but think is the general best practice to avoid situations like that recurring Android crash in flagship.

narin commented:
- Deleted useA11yListPosition hook and reverted to using `listPosition` translation directly since we're now using `aria-valuetext`
- Added unit tests to check for heading role and accessibility value
- Fix bug where Talkback wasn't reading a11y labels for Label and Description but combining aria-labels and setting on Pressable

TimRoe commented on packages/components/src/components/Checkbox/Checkbox.tsx:
```diff
@@ -20,7 +20,8 @@ import {
 import { Spacer } from '../Spacer/Spacer'
-import { useTheme } from '../../utils'
+import { getA11yLabel, useTheme } from '../../utils'
+import { useTranslation } from 'react-i18next'
```
Nitpick: move up to library imports.

TimRoe commented on packages/components/src/components/CheckboxGroup/CheckboxGroup.tsx:
```diff
@@ -11,7 +11,8 @@ import {
 import { Spacer } from '../Spacer/Spacer'
-import { useA11yListPosition, useTheme } from '../../utils'
+import { useTheme } from '../../utils'
+import { useTranslation } from 'react-i18next'
```
Same nitpick as other file

TimRoe commented on packages/components/src/components/SegmentedControl/SegmentedControl.tsx:
```diff
@@ -4,11 +4,8 @@ import React, { FC, useEffect } from 'react'
-  useTheme,
-} from '../../utils'
+import { PressableOpacityStyle, useTheme } from '../../utils'
+import { useTranslation } from 'react-i18next'
```
Same nitpick as other files

TimRoe commented on packages/components/src/components/SegmentedControl/SegmentedControl.tsx:
```diff
@@ -103,6 +97,11 @@ export const SegmentedControl: FC<SegmentedControlProps> = ({
       textAlign: 'center',
     }
 
+    const a11yListPosition = t('listPosition', {
```
Nitpicky, but old position of setting the const alongside the a11yLabel made more sense readability-wise

TimRoe reviewed (approved):
Approved with a few lingering nitpicks about import/const positioning.
